src/sqlfluff/core/parser/segments/base.py

```diff
@@ -84,11 +84,13 @@ class BaseSegment:
     allow_empty = False
     # What other kwargs need to be copied when applying fixes.
     additional_kwargs: List[str] = []
+    # This typing isnt really true as we set the pos_marker to None in places
+    pos_marker: PositionMarker
 
     def __init__(
         self,
         segments,
-        pos_marker=None,
+        pos_marker: PositionMarker = None,
```

Comment:
Having a default value of `None` means it must be `Optional`.
```suggestion
        pos_marker: Optional[PositionMarker] = None,
```